### DISCUSSION TITLE
DashPanel flaky with Chrome

### DISCUSSION THREAD

**Discussion by jmillerhyetech** [tagyoureit/nodejs-poolController#791] [Q&A]
For the past week or so I have had issues using Chrome on the poolcontroller dashpanel. I have problems with both Chrome on my mobile device as well as my Mac. I've tried clearing cache and using incognito with no change in behavior.
The problem is I used to be able to change settings for say the chlorinator, or the heater/solar temp by clicking on the % output or the set temperature, etc. However, clicking this area does nothing. I am also not able to click the hamburger menu, but the settings menu at the top right works OK. 
Using Safari browser works OK.

am running poolcontroller ver 7.7.0
I just did a git pull and a npm i for dashpanel and have the same behavior. 
let me know if I can provide any additional information.

**Comment by rstrouse** [comment]
A recent update to chrome hijacks the popovers.   You need to update to version 8.x for dashPanel and njsPC.

**Reply by jmillerhyetech** [reply]
Thanks for the reply. I was hoping just updating dashpanel would do the trick. One question before pulling poolcontroller 8 version.
Back in June last year you helped me with using Easytouch and Nixie as primary controllers on the same RS485 bus. (see #483)
If I pull the new version will this continue to work?

**Reply by rstrouse** [reply]
Yes this will continue to work.  You will probably need to upgrade to node 18.  Are you running pm2?

**Reply by rstrouse** [reply]
To update when you are running pm2 and have used nvm.
`pm2 stop all`
`nvm install 18`
`nvm alias default 18`
`npm update -g`
`npm install -g pm2`
`cd ~/nodejs-poolController`
`git pull`
`npm i`
`cd ~/nodejs-poolController-dashPanel`
`git pull`
`npm i`
`pm2 start ecosystem.config.js`
`pm2 save`
`pm2 startup`
-- After this command copy the command text provided on the screen and run it.  The instructions will be in the prompt.

**Reply by Telemachus1977** [reply]
Upgrading booth fixed my menu popups on my PC, but mobile Chrome still not working...

**Comment by jmillerhyetech** [comment]
Yes, I updated to node18 and am running pm2 as well.
-j

**Reply by rstrouse** [reply]
Did this fix your chrome issues?

**Reply by jmillerhyetech** [reply]
I haven't had a chance to test yet. I will let you know.

**Reply by rstrouse** [reply]
Make sure you hard refresh CTRL+F5 or clear the browser cache.

**Reply by jmillerhyetech** [reply]
Sorry for the delay. I was on vacation last week. I just upgraded as directed above. Looks like the RS485 isn't initializing now. I am getting the following stacktrace when making changes in the gui.
```

Error: OutboundMessageError
MessageComms port 0 is not open. Message aborted: 165,33,16,33,134,2,4,1,1,132
ApiError: Comms port 0 is not open. Message aborted: 165,33,16,33,134,2,4,1,1,132 
    at RS485Port.processOutboundPackets (/home/pi/nodejs-poolController/controller/comms/Comms.ts:616:29)
    at RS485Port.processPackets (/home/pi/nodejs-poolController/controller/comms/Comms.ts:578:14)
    at Immediate._onImmediate (/home/pi/nodejs-poolController/controller/comms/Comms.ts:542:81)
    at processImmediate (node:internal/timers:476:21)
```

**Reply by rstrouse** [reply]
Run the following commands.

`pm2 stop all`
`cd ~/nodejs-poolController`
`npm rebuild`
`pm2 restart all`

**Reply by jmillerhyetech** [reply]
Same behavior

**Reply by rstrouse** [reply]
Take a screenshot of your RS485 port configuration in dashPanel.

**Reply by jmillerhyetech** [reply]
<img width="631" alt="image" src="https://github.com/tagyoureit/nodejs-poolController/assets/55600904/cef59623-4ef5-493c-804a-4b4867dc0ecb">

**Reply by rock-crusher** [reply]
Hi jmillerhyetech,
My Primary Port is listed as:  /dev/ttyS0
Hope this helps

**Reply by jmillerhyetech** [reply]
Mine is definitely ttyUSB0
```
pi@raspberrypi:/var/log $ tail syslog
Jun 28 12:45:04 raspberrypi kernel: [24291.874943] usb 1-1.2: Detected FT232R
Jun 28 12:45:04 raspberrypi kernel: [24291.877837] usb 1-1.2: FTDI USB Serial Device converter now attached to ttyUSB0
Jun 28 12:45:04 raspberrypi mtp-probe: checking bus 1, device 6: "/sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2"
Jun 28 12:45:04 raspberrypi mtp-probe: bus: 1, device: 6 was not an MTP device
Jun 28 12:45:04 raspberrypi mtp-probe: checking bus 1, device 6: "/sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2"
Jun 28 12:45:04 raspberrypi mtp-probe: bus: 1, device: 6 was not an MTP device
Jun 28 12:45:07 raspberrypi ModemManager[545]: <info>  [base-manager] couldn't check support for device '/sys/devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2': not supported by any plugin
Jun 28 12:49:35 raspberrypi systemd[1]: session-13.scope: Succeeded.
Jun 28 12:49:35 raspberrypi systemd[1]: session-13.scope: Consumed 1.961s CPU time.
Jun 28 12:51:10 raspberrypi systemd[1]: Started Session 14 of user pi.
```
this is all I get though when trying to look for the RS485 data
```
pi@raspberrypi:/var/log $ od -x < /dev/ttyUSB0
0000000 ff00 ffff ffff ffff 00ff a5ff 0f02 0210
0000020 0a1d 02ff 000a
0000025
pi@raspberrypi:
```

not sure what is going on here. It was working prior to upgrading npm and poolcontroller.

**Reply by rock-crusher** [reply]
Try changing it to /dev/ttyS0 and see what happens, also don’t forget to hit the save port button at the bottom.
When mine was defaulting to /dev/ttyUSB0 it didn’t work for me.
Hopefully rstrouse will chime in here and save the day!

**Reply by rstrouse** [reply]
What RS485 adapter do you have?

**Comment by jmillerhyetech** [comment]
I have several of the Moyina adapters recommended in the Wiki. I did try swapping it out though and that made no difference. I fussed with it last night a little more, and I elected just to rebuild the Pi from scratch. Works perfectly now. I do still have the microSD, so if you would like some files to look over I can get them. Otherwise you can probably mark as resolved.
